From c4beac7712f993369687f62f4c2c2b2cda256a56 Mon Sep 17 00:00:00 2001
From: Richard Laager <rlaager@wiktel.com>
Date: Mon, 12 Dec 2016 01:49:26 -0600
Subject: [PATCH 2/5] Handle gpsd returning None as the device

For whatever reason, the python bindings for gpsd return None as the
device.  This handles that case gracefully, by showing "TDOP" and "nSat"
instead of "None TDOP" and "None nSat".
---
 ntpclients/ntpviz | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

--- a/ntpclients/ntpviz
+++ b/ntpclients/ntpviz
@@ -604,9 +604,13 @@
 """ % out
 
         for key in gpslist:
+            if key is None:
+                key = ""
+            else:
+                key = key + " "
             plot_template += """\
-'-' using 1:2 title '%(key)s TDOP' with line ls 1, \\
-'-' using 1:3 title '%(key)s nSat' with line ls 2 axis x1y2, \\
+'-' using 1:2 title '%(key)sTDOP' with line ls 1, \\
+'-' using 1:3 title '%(key)snSat' with line ls 2 axis x1y2, \\
 """ % locals()
 
         # strip the trailing ", \\n"
